### PR TITLE
[lua] Export cmake targets, rewrite CMakeLists and delete usage

### LINKED ISSUE #15497: [lua] cmake can not return on linux, while compiling, the cpu usage is 100%, and the task will never end
**Describe the bug**
when i use lua
cmake can not return on linux, while compiling, the cpu usage is 100%, and the task will never end

**Environment**
- OS: Ubuntu20
- Compiler: cmake 3.16

**To Reproduce**
vcpkg install lua
cmake .. "-DCMAKE_TOOLCHAIN_FILE=~/.vcpkg/scripts/buildsystems/vcpkg.cmake"

**Expected behavior**
compile complete successfully

**Failure logs** 
-not any logs

**Additional context**
the same code can run perfectly on win10.

```
# CMakeLists.txt
cmake_minimum_required(VERSION 3.0)
project(test)

include(FindLua)
find_package(Lua REQUIRED)

add_executable(main main.cpp)

target_link_libraries(main PRIVATE ${LUA_LIBRARIES})

```

### PR DESCRIPTION
**Pull request by JackBoosY**
Since `FindLua.cmake` sets the default search path, and the header file is included in lua-version by default, it is impossible to find the lua generated by vcpkg correctly, export the targets and modify the cmake wrapper to adapt to the search method of the old version.

Fixes #15497.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@williamxin Can you please test this PR?

Thnaks.

**Comment by williamxin** [comment]
> @williamxin Can you please test this PR?
> 
> Thanks.

thank you!
i am working on it, but there is another issue in vcpkg,  everything works perfect on windows, but bootstrap.sh failed on my mac

**Comment by williamxin** [comment]
> @williamxin Can you please test this PR?
> 
> Thanks.

now cmake can finish on my centos,  the problem has been solved, thank you

**Review by NancyLi1013 — APPROVED** [review]

File: ports/lua/vcpkg-cmake-wrapper.cmake.in [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+foreach(ARG ${ARGS})
+    if (ARG STREQUAL "REQUIRED")
+        set(REQUIRES "REQUIRED")
+        break()
+    endif()
+endforeach()
+
+_find_package(unofficial-lua CONFIG ${REQUIRES})
+
+set(LUA_LIBRARIES lua)
```
Comment (ras0219-msft) [inline ports/lua/vcpkg-cmake-wrapper.cmake.in:11]: This file should also set `LUA_INCLUDE_DIR` to match FindLua.cmake.

File: ports/lua/usage [hunk 2]
```diff
@@ -1,9 +0,0 @@
-Use this package via the module FindLua that comes with CMake. To use in your CMakeLists.txt:
```
Comment (ras0219-msft) [inline ports/lua/usage:1]: We should continue providing usage instructions that suggest the user to use the official targets:
```
find_package(Lua REQUIRED)
target_include_directories(main PRIVATE ${LUA_INCLUDE_DIR})
target_link_libraries(main PRIVATE ${LUA_LIBRARIES})
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/lua/usage (hunk 2 above)
Comment (JackBoosY) [reply ports/lua/usage:1]: Done.

File: ports/lua/vcpkg-cmake-wrapper.cmake.in (hunk 1 above)
Comment (JackBoosY) [reply ports/lua/vcpkg-cmake-wrapper.cmake.in:15]: Done.

**Comment by JackBoosY** [comment]
@ras0219-msft Do you have any other questions?

**Comment by pkeir** [comment]
I tried this PR on Ubuntu 20.10 and still face a problem: see [here](https://github.com/microsoft/vcpkg/issues/16438#issuecomment-789577780).

**Comment by JackBoosY** [comment]
The tensorflow-cc regression will be resolved by #19165.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**